cmd/entc/entc.go

```diff
@@ -115,6 +115,43 @@ func main() {
 			cmd.Flags().StringSliceVarP(&template, "template", "", nil, "external templates to execute")
 			return cmd
 		}(),
+		&cobra.Command{
+			Use:   "clean [flags] path",
+			Short: "delete generated go code",
+			Example: examples(
+				"entc clean ./ent",
+			),
+			Args: cobra.ExactArgs(1),
+			Run: func(cmd *cobra.Command, path []string) {
+				cleanup := make([]string, 0)
+				cleanup = append(cleanup, filepath.Join(path[0], "migrate"))
+				cleanup = append(cleanup, filepath.Join(path[0], "predicate"))
+				cleanup = append(cleanup, filepath.Join(path[0], "client.go"))
+				cleanup = append(cleanup, filepath.Join(path[0], "config.go"))
+				cleanup = append(cleanup, filepath.Join(path[0], "context.go"))
+				cleanup = append(cleanup, filepath.Join(path[0], "ent.go"))
+				cleanup = append(cleanup, filepath.Join(path[0], "tx.go"))
```

Comment:
Let's change this to `filepath.Walk` and remove all files that were generated by `entc` (`go generate` signature matching).
It will be easier to maintain, and it will support deleting files that were generated with external templates as well. 

Thanks

Reply:
okay. i didn't thought about files generated by external  templates. i start working on that.

Reply:
done

Reply:
Please address this to all files.

You should use `filepath.Walk` under the given directory and check that the file was generated by `entc` (on package comment).

Reply:
what do you mean with "on package comment"? checking if the file has a line `// Code generated by entc, DO NOT EDIT.` ?

is it not better to use the `Templates` and `GraphTemplates ` array from the  `entc/gen/template.go` source as the list of files need to be deleted?

https://github.com/facebookincubator/ent/blob/dc222c0b15343920adac37c2c60d292514b50708/entc/gen/template.go#L39-L106

Reply:
>  checking if the file has a line // Code generated by entc, DO NOT EDIT. ?

Yes. 

> is it not better to use the Templates and GraphTemplates array from the entc/gen/template.go source as the list of files need to be deleted?

Custom templates generate additional files (and directories).

Reply:
> checking if the file has a line // Code generated by entc, DO NOT EDIT. ?

i'm not sure if this works well because if a custom template do not write that line it will not be deleted. for the build in templates it is more expensive to open and check the file. we know the files the `generate` command has created.

my initial strategy was to read the `ent/schema/` directory, find all files with a `.go` extension and then checking the `ent/` directory for files that start with that name. like it is defined at the Templates array (`%s_create.go`, `%s_update.go`, etc.)


a strategy for custom templates could be to add a `templates` flag, similar to the flag available at the `generate` command to know the files we need to delete.

Reply:
Sorry, I've missed this comment. 

> i'm not sure if this works well because if a custom template do not write that line it will not be deleted. 

I'm fine if it won't work for these case - it's Go convention and just needs to be documented. All the external templates we provide to users have the proper headers. 

> we know the files the generate command has created.
>
> my initial strategy was to read the ent/schema/ directory, find all files with a .go extension and then checking the ent/ directory for files that start with that name. like it is defined at the Templates array (%s_create.go, %s_update.go, etc.)

Schema files can have more than one definition - valid case. In this case, you'll need to read the schema files, find all declarations and then find their generated files.